pkg/tun/tcp/handler.go

```diff
@@ -303,12 +313,19 @@ func (h *handler) sendSyn(ctx context.Context, ackNumber uint32, ack bool) {
 
 // writeToTunLoop sends the packages read from the fromMgr channel to the TUN device
 func (h *handler) writeToTunLoop(ctx context.Context) {
+	defer close(h.readyToFin)
+	timer := time.NewTimer(100 * time.Millisecond)
 	for {
 		var cm connpool.Message
 		select {
 		case <-ctx.Done():
 			return
 		case cm = <-h.fromMgr:
+		case <-timer.C:
```

Comment:
From the looks of it, this timer will only make difference on very short-lived connections. A timer will only fire once. Perhaps you intended to use a `Ticker` or: `case <-time.After(100 * time.Milliecond):`?

Reply:
Ah yes this should definitely be a `Ticker`